samples/nrf9160/modem_shell/src/link/link.c

```diff
@@ -309,6 +320,16 @@ void link_ind_handler(const struct lte_lc_evt *const evt)
 				cells.neighbor_cells[i].earfcn,
 				cells.neighbor_cells[i].time_diff);
 		}
+
+		ncellmeas_work_data.mode = ncellmeas_mode;
+		ncellmeas_work_data.search_type = ncellmeas_search_type;
+		ncellmeas_work_data.periodic_interval = ncellmeas_periodic_interval;
+		if (ncellmeas_mode == LINK_NCELLMEAS_MODE_CONTINUOUS &&
+		    ncellmeas_periodic_interval) {
+			/* Interval was given for continuous mode */
+			k_work_schedule(&ncellmeas_work_data.work,
```

Comment:
Just wondering if it would be better to use `k_work_reschedule()` here instead. That would affect the case where interval mode is enabled and a cell change triggers neighbor cell measurement in between. Currently, the next scheduled measurement would happen `ncellmeas_periodic_interval` seconds from the previous measurement triggered by the interval, but would it make sense to use `k_work_reschedule()` so that the next scheduled measurement would happen `ncellmeas_periodic_interval` seconds after the one triggered by the cell change instead?

Reply:
Yes, agree, a good point, thanks! Changed.